code/datums/diseases/wizarditis.dm

```diff
@@ -87,32 +87,36 @@ STI KALY - blind
 
 
 /datum/disease/wizarditis/proc/teleport()
-	var/list/theareas = get_areas_in_range(80, affected_mob)
-	for(var/area/space/S in theareas)
-		theareas -= S
+	var/list/possible_areas = get_areas_in_range(80, affected_mob)
+	for(var/area/space/S in possible_areas)
+		possible_areas -= S
 
-	if(!theareas||!theareas.len)
+	if(!length(possible_areas))
 		return
 
-	var/area/thearea = pick(theareas)
+	var/area/chosen_area = pick(possible_areas)
 
-	var/list/L = list()
-	for(var/turf/T in get_area_turfs(thearea.type))
-		if(T.z != affected_mob.z) continue
-		if(T.name == "space") continue
+	var/list/teleport_turfs = list()
+	for(var/turf/T in get_area_turfs(chosen_area.type))
+		if(!is_teleport_allowed(T.z))
+			break
+		if(T.z != affected_mob.z)
+			continue
```

Comment:
Shouldn't the first loop be `continue`? And if it needs to exit early, shoulnd't the second loop also do that since its relating to Z checks?

Reply:
just added them together and made it break